app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -333,12 +334,30 @@ def redirect_url
       if !mobile? && TwoFactorAuthentication::PivCacPolicy.new(current_user).enabled?
         login_two_factor_piv_cac_url
       elsif TwoFactorAuthentication::WebauthnPolicy.new(current_user).enabled?
-        login_two_factor_webauthn_url(webauthn_params)
+        webauthn_url
```

Comment:
I'd sorta wonder if this should be a `before_action` consideration of the WebAuthn controller, to avoid this generic MFA controller having to be overly concerned with specifics of individual methods. What do you think?

Reply:
I kind of dont want it to be in the before action because then a user would never be able to get to that page if their device is "ineligible" but the way webauthn works, users may be using another device to access it via an ineligible device. For example QR code scanning with my phone while logging in with desktop.

Reply:
I think a before action may be cleaner. updating.